raft.go

```diff
@@ -792,6 +799,11 @@ func (r *Raft) runLeader() {
 			select {
 			case notify <- false:
 			case <-r.shutdownCh:
+				// On shutdown, make a best effort but do not block
```

Comment:
Its a race condition in one of the unit tests, basically during a shutdown we may fail to push to notify channel, so this tries to fix it without blocking.